python/fledge/plugins/south/person_detection/person_detection.py

```diff
@@ -367,19 +404,30 @@ def __init__(self, handle):
         _ = self.inference.get_interpreter(model, enable_tpu,
                                            labels, self.min_conf_threshold)
 
-        source = int(handle['camera_id']['value'])
-
         if handle['enable_window']['value'] == 'true' and not FrameProcessor.check_background():
             self.enable_window = True
         else:
             self.enable_window = False
 
-        # Initialize the stream object and start the thread that keeps on reading frames
-        # This thread is independent of the Camera Processing Thread
-        self.videostream, is_camera_functional = VideoStream(resolution=(self.camera_width,
-                                                                         self.camera_height),
-                                                             source=source).start()
-        self.is_camera_functional = is_camera_functional
+        if handle['source']['value'] == 'camera':
+            camera_id = int(handle['camera_id']['value'])
+            # Initialize the stream object and start the thread that keeps on reading frames
+            # This thread is independent of the Camera Processing Thread
+            self.videostream, is_camera_functional = VideoStream(resolution=(self.camera_width,
```

Comment:
If there are any errors in setting-up the video input, exact error must be logged to syslog.

Reply:
**is_camera_functional** will be False in that case.